crawl (1:4.0.0beta26-4.kon.1) local; urgency=medium

  * Local version!
  * source/dungeon.cc (box_room_doors): Give up if there are not enough
    good spots, instead of falling in an infinite loop.

 -- Kalle Olavi Niemitalo <kon@iki.fi>  Sat, 18 Jun 2005 10:07:51 +0300

--- crawl-4.0.0beta26.orig/source/dungeon.cc
+++ crawl-4.0.0beta26/source/dungeon.cc
@@ -5,7 +5,8 @@
  *
  *  Change History (most recent first):
  *
- *
+ *   <10>    18-Jun-2005 KON    box_room_doors gives up if there are not
+ *                              enough good spots.
  *   <9>     07-Aug-2001 MV     clean up of give_item; distribution of
  *                              wands, potions and scrolls
  *                              underground rivers and lakes
@@ -7538,6 +7539,14 @@
         return doors_placed;
     }
 
+    // Avoid an infinite loop if there are not enough good spots. --KON
+    j = 0;
+    for (i=0; i<spot_count; i++)
+        if (good_doors[i] == 1)
+            j++;
+    if (new_doors > j)
+        new_doors = j;
+
     while(new_doors > 0 && spot_count > 0)
     {
         spot = random2(spot_count);
